packages/jest-config/src/normalize.js

```diff
@@ -236,6 +236,19 @@ const normalizeMissingOptions = (options: InitialOptions): InitialOptions => {
       .digest('hex');
   }
 
+  if (Array.isArray(options.projects)) {
+    options.projects = options.projects.map((project, index) => {
+      if (typeof project !== 'string' && !project.name) {
+        project.name = crypto
+          .createHash('md5')
+          .update(project.rootDir || `${options.rootDir}:${index}`)
```

Comment:
won't this give the same hash if multiple projects have the same `rootDir`? Also, if it's falsy, you'll get `undefined:0` or something like it?

Reply:
@cbelsole thoughts on the above?

Reply:
I changed it around a bit to keep a cache of all root directories to avoid duplicates and beef up the tests. Since `options.rootDir` is required we should not get undefined, but even if we do it will be a valid unique name.